modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/CalciteQueryProcessorTest.java

```diff
@@ -1216,6 +1216,41 @@ public void testInfixTypeCast() {
         assertThat(qCur.fieldsMeta().get(3).fieldTypeName(), equalTo(String.class.getName()));
     }
 
+    /** Quantified predicates test. */
+    @Test
+    public void quantifiedCompTest() throws InterruptedException {
```

Comment:
We need to add more test to get sure that correlated queries are planned and executed properly. So let's provide a bunch of planner + integration tests to verify:

- correlated queries in SELECT list (simple projection, aggregates)
- correlated queries as part of predicate (=, between, some/any, all, in, exists)
- correlates in table functions (currently presented by `system_range` function)

It would be nice to cover cases with multiple correlated variables.